demo/server/src/modules/datasync/MongoDataHandler.ts

```diff
@@ -27,8 +28,9 @@ export class GlobalMongoDataHandler {
   public setupHandleList() {
     const self = this;
     sync.globalHandleList(function(datasetId, queryParams, metadata, cb) {
+      logger.debug('GlobalHandleList called', datasetId, queryParams);
       queryParams = queryParams || {};
-      const resultPromise = this.db.collection(datasetId).find(queryParams);
+      const resultPromise = self.db.collection(datasetId).find(queryParams);
```

Comment:
optional: can remove the need for `self` if you use arrow functions instead of closures